src/lib/service/learnerRecordAPI/moduleRecord/patchFactory.ts

```diff
@@ -0,0 +1,67 @@
+import * as moment from 'moment'
+
+import { JsonPatch } from '../../shared/models/JsonPatch'
+import { RecordState } from '../models/record'
+import { BookingStatus, ModuleRecordResult } from './models/moduleRecord'
+
+export function clearBookingStatus() {
+	return JsonPatch.removePatch('bookingStatus')
+}
+
+export function clearResult() {
+	return JsonPatch.removePatch('result')
+}
+
+export function clearScore() {
+	return JsonPatch.removePatch('score')
+}
+
+export function clearCompletionDate() {
+	return JsonPatch.removePatch('completionDate')
+}
+
+export function setScore(score?: string) {
+	return JsonPatch.replacePatch('score', score)
+}
+
+export function setRated(rated: boolean) {
+	return JsonPatch.replacePatch('rated', rated.toString())
+}
+
+export function setResult(result?: ModuleRecordResult) {
+	return JsonPatch.replacePatch('result', result)
+}
+
+export function setEventId(eventId: string) {
+	return JsonPatch.replacePatch('eventId', eventId)
+}
+
+export function setEventDate(eventDate?: Date) {
+	return setDate('eventDate', eventDate)
+}
+
+function setDate(key: string, date?: Date) {
+	let convertedDate
+	if (date) {
+		convertedDate = moment(date).format('YYYY-MM-DDTHH:mm:ss')
+	} else {
+		convertedDate = moment(new Date()).format('YYYY-MM-DDTHH:mm:ss')
```

Comment:
Setting the current date when the `date` argument is not specified seems like a bit of a ["side effect"](https://dipta007.com/clean-code-functions/#no-side-effects) and not clear from the function definition.